exporter/alertmanagerexporter/factory.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package alertmanagerexporter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/configopaque"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr   = "alertmanager"
```

Comment:
When you have the go:generate directive in place, you'll have an internal metadata package that holds this information already, based on your metadata.yaml. Note how your stability on the next line differs from the one declared in the metadata.yaml.